test/lib/mayaUsd/render/mayaToHydra/cpp/testIsolateSelectMayaSelectionHighlighting.py

```diff
@@ -0,0 +1,181 @@
+        # Clean out the scene to allow all plugins to unload cleanly.
+        cmds.file(new=True, force=True)
+        for p in reversed(cls._pluginsToUnload):
+            cmds.unloadPlugin(p)
+
+    def setupScene(self):
+
+        cmds.polyTorus()
+        cmds.polySphere()
+        cmds.move(2, 0, 0, r=True)
+        cmds.polyCube()
+        cmds.move(-2, 0, 0, r=True)
+        cmds.group('pSphere1', 'pCube1')
+        cmds.move(0, 0, 2, r=True)
+
+        cmds.refresh()
+
+        scene = [
+            # Maya objects
+            '|pTorus1', 
+            '|pTorus1|pTorusShape1', 
+            '|group1',
+            '|group1|pSphere1',
+            '|group1|pSphere1|pSphereShape1',
+            '|group1|pCube1',
+            '|group1|pCube1|pCubeShape1']
+
+        return scene
+
+    def assertVisible(self, visible):
+        for v in visible:
+            self.trace("Testing %s for visibility\n" % v)
+            cmds.mayaHydraCppTest(v, f="TestHydraPrim.isVisible")
+
+    def assertNotVisible(self, notVisible):
+        for nv in notVisible:
+            self.trace("Testing %s for invisibility\n" % nv)
+            cmds.mayaHydraCppTest(nv, f="TestHydraPrim.notVisible")
+
+    def assertVisibility(self, visible, notVisible):
+        self.assertVisible(visible)
+        self.assertNotVisible(notVisible)
+
+    def assertIsolateSelect(self, modelPanel, visible, scene):
+        cmds.select(*visible)
+        cmds.editor(modelPanel, edit=True, updateMainConnection=True)
+        cmds.isolateSelect(modelPanel, loadSelected=True)
+
+        notVisible = scene.copy()
+
+        for p in visible:
+            notVisible.remove(p)
+
+        cmds.refresh()
+
+        self.assertVisibility(visible, notVisible)
+
+    def test_isolateSelectMayaSelectionHighlighting(self):
+
+        scene = self.setupScene()
+
+        cmds.select(clear=True)
+
+        # Isolate select not turned on, everything visible.
+        visible = scene
+        notVisible = []
+
+        cmds.refresh()
+
+        self.assertVisibility(visible, notVisible)
+
+        # Turn isolate select on, nothing selected, nothing visible.
+        visible = []
+        notVisible = scene
+
+        modelPanel = 'modelPanel4'
+        enableIsolateSelect(modelPanel)
+        
+        cmds.refresh()
+
+        self.assertVisibility(visible, notVisible)
+
+        # Now select Maya objects and check both object and selection
+        # highlighting visibility.
+        cmds.select('|pTorus1')
+        cmds.editor(modelPanel, edit=True, updateMainConnection=True)
+        cmds.isolateSelect(modelPanel, loadSelected=True)
+
+        cmds.refresh()
+
+        cmds.mayaHydraCppTest('/MayaHydraViewportRenderer/rprims/pTorus1',
+                              f='TestPrimPath.isVisible');
+        cmds.mayaHydraCppTest('/MayaHydraViewportRenderer/rprims/Lighted/pTorus1',
+                              f='TestPrimPath.isVisible');
```

Comment:
Both the selection highlighting and the real geometry are visible under isolate select.